applications/HDF5Application/custom_io/hdf5_connectivities_data.cpp

```diff
@@ -0,0 +1,191 @@
+#include "custom_io/hdf5_connectivities_data.h"
+
+#include "includes/kratos_components.h"
+#include "utilities/openmp_utils.h"
+#include "utilities/compare_elements_and_conditions_utility.h"
+
+namespace Kratos
+{
+namespace HDF5
+{
+namespace Internals
+{
+void ConnectivitiesData::ReadData(File& rFile, const std::string& rPath, unsigned StartIndex, unsigned BlockSize)
+{
+    KRATOS_TRY;
+    Clear();
+    rFile.ReadAttribute(rPath, "Name", mName);
```

Comment:
one question: what happens if you have to read multiple elements types?

Reply:
Connectivities data only represents a single element type. If the model part has more than one element or condition type, then a new connectivities data object is created for each type. Each instance is then responsible for writing its data to the file with the label given by the element or condition type. An example (with only one element type) is shown in the figure. 
![connectivities](https://user-images.githubusercontent.com/22290001/36165969-7d3329c6-10f1-11e8-9a8b-9b25afb225b2.png)

Reply:
what about submodelparts?

Reply:
Sub model parts are on the todo list, but not there yet.